lib/spack/spack/new_installer.py

```diff
@@ -623,22 +627,45 @@ def toggle(self) -> None:
                 pass
             self.tracked_build_id = ""
 
+    def search_input(self, input: str) -> None:
+        """Handle keyboard input when in search mode"""
+        if input in ("\r", "\n"):
+            self.next(1)
+        elif input == "\x1b":  # Escape
+            self.search_mode = False
+            self.search_term = ""
+            self.dirty = True
+        elif input in ("\x7f", "\b"):  # Backspace
+            self.search_term = self.search_term[:-1]
+            self.dirty = True
+        elif input.isprintable():
+            self.search_term += input
+            self.dirty = True
+
+    def enter_search(self) -> None:
+        self.search_mode = True
+        self.dirty = True
+
+    def _matches_search(self, build: BuildInfo) -> bool:
+        """Return true when the selected build matches the search term"""
+        return self.search_term in build.name or build.hash.startswith(self.search_term)
+
     def _get_next(self, direction: int) -> Optional[str]:
-        """Returns the next or previous unfinished build ID, or None if none found.
-        Direction should be 1 for next, -1 for previous."""
-        build_ids = list(self.builds)
+        """Returns the next or previous unfinished build ID matching the search term, or None if
+        none found. Direction should be 1 for next, -1 for previous."""
+        matching = [
+            build_id
+            for build_id, build in self.builds.items()
+            if build.finished_time is None and self._matches_search(build)
+        ]
+        if not matching:
+            return None
```

Comment:
Would be easier to read with a comment noting that the search term is `""` by default, which means it matches any build.

Reply:
I moved the comment into the function, and renamed the function to something that does not suggest that a search term must be present.